GabbyTam_Fatima_functionpractice.js

```diff
@@ -0,0 +1,238 @@
+
+function getTwoLengths(first, second) {
+  let arrayNum= [];
+  arrayNum.push(first.length);
+  arrayNum.push(second.length);
+  return arrayNum;
+}
+//but why is in not .length-1?? 
+//answer: -1 and +1 comes into play when you are using for loops, declaring your variable at 0 or at 1 will make a difference as to what add on youll need to .length property 
+// .length counts string and arrays starting at index 0
+ 
+console.log(getTwoLengths("Hank", "Hippopopalous"));
+
+
+//Part 1: 4 Write a function getMultipleLengths that accepts a single parameter as an argument: an array of strings. The function should return an array of numbers where each number is the length of the corresponding string.
+
+const getMultipleLengths = (array) => {
+  let arrayOfNum= [];
+  for(let i=0; i<= array.length-1; i++){
+    arrayOfNum.push(array[i].length); 
+  }
+  return arrayOfNum;
+  //console.log(arrayOfNum);
+}
+
+console.log(getMultipleLengths(["hello", "what", "is", "up", "dude"]));
+
+//Part 2: 1 Define a function maxOfThree that takes three numbers as arguments and returns the largest of them. If all numbers are the same, it doesn't matter which one is returned. If the two largest numbers are the same, one of them should be returned.
+
+
+/*
+first attempt: 
+
+ const maxOfThree = (num1, num2, num3) => {
+  if(num1> num2 && num1> num3){
+    return num1;
+  }
+  if(num2> num1 && num2> num3){
+    return num2;
+  }
+  if(num3> num1 && num3 > num2){
+    return num3;
+  }
+  if(num1== num2== num3){
+    return num1;
+  }
+}
+
+console.log(maxOfThree(6, 9, 1));
+
+*/
+
+//attempt after googling 
+const maxOfThree = (num1, num2, num3) => {
+  let max= Math.max(num1,num2,num3);
+  return max;
+}
+
+console.log(maxOfThree(6, 9, 1));
+
+//Part 2: 2 Write a function printLongestWord that accepts a single argument, an array of strings. The method should return the longest word in the array. In case of a tie, the method should return the word that appears first in the array.
+
+/* This doesnt work because of the use of return, return stops/breaks the code
+
+
+function printLongestWord(arr){
+  for(let i=0; i<arr.length-1;i++){
+    if(arr[i].length> arr[i+1].length){
+      return arr[i];
+    } else {
+      return arr[i+1];
+    }
+  }
+}
+
+console.log(printLongestWord(["BoJack", "Princess", "Diane", "a", "Max", "Peanutbutter", "big", "blob"]));
+*/
+
+//this works but only for this set of strings, if the longest word comes before peanutbutter it is overwritten by peanutbutter in later computation 
+
+function printLongestWord(arr){
+  let longest= '';
+  for(let i=0; i<arr.length-1; i++){
+    if(arr[i].length> arr[i+1].length){
+
+      longest= arr[i];
+    } //else {
+    //continue;
+   // }
+  } 
+  return longest;
+}
+
+console.log(printLongestWord(["BoJack", "Princess", "Diane", "a", "Max", "Peanutbutter", "big", "blob"]));
```

Comment:
you're really close here! start with the assumption that the first element is the longest, and then use the loop to compare the length of the value currently saved as the longest with the next element in the array. You should also switch your `>` to `<` in line 130. It should look like this:

```javascript
function printLongestWord(arr){
    let longest = arr[0];
    for(let i = 0; i < arr.length - 1; i++){
        if(longest.length < arr[i+1].length){
            longest = arr[i + 1];
            console.log(longest)
        }
    } 
    console.log("this is the longest:")
    return longest;
  }
```

[also, watch your indentation!]